polkadot/runtime/parachains/src/paras_inherent/mod.rs

```diff
@@ -1404,10 +1513,9 @@ fn map_candidates_to_cores<T: configuration::Config + scheduler::Config + inclus
 						);
 						break;
 					}
-					let maybe_injected_core_index: Option<CoreIndex> =
-						get_injected_core_index::<T>(allowed_relay_parents, &candidate);
 
-					if let Some(core_index) = maybe_injected_core_index {
+					if let Some(core_index) = get_core_index::<T>(allowed_relay_parents, &candidate)
```

Comment:
get_core_index should be parametrised with `core_index_enabled` and only use the injected core index if the feature is enabled.
the `} else if scheduled_cores.len() >= 1 && core_index_enabled {` should then be relaxed.

actually, get_core_index should also look into the core index field of the descriptor only if the `CandidateReceiptV2` feature is enabled

Reply:
> get_core_index should be parametrised with `core_index_enabled` and only use the injected core index if the feature is enabled. the `} else if scheduled_cores.len() >= 1 && core_index_enabled {` should then be relaxed.
> 

`core_index_enabled` is currently enabled  on all chains and in the past the significance of this was enablement of elastic scaling. I  don't think we need to pass it in as a parameter, we can just assume the core index is injected as fallback in case the descriptor is not v2.

> actually, get_core_index should also look into the core index field of the descriptor only if the `CandidateReceiptV2` feature is enabled

`map_candidates_to_cores` happens after backed candidate sanitization. So if there are a any v2 descriptors at this point, it means the feature is enabled.

Reply:
I was eager to get rid of the MVP flag, but this is not the place. We need to remove the node usage of that and assume we always inject the core in the `BackedCandidate` struct. 

What you said makes sense.

Reply:
I've refactored `map_candidates_to_cores` to be simpler to read and reduced the number of checks.